Library/Homebrew/cask/cmd/upgrade.rb

```diff
@@ -17,35 +18,76 @@ def initialize(*)
       end
 
       def run
-        outdated_casks = casks(alternative: lambda {
+        self.class.upgrade_casks(
+          *casks,
+          force:          force?,
+          greedy:         greedy?,
+          dry_run:        dry_run?,
+          binaries:       binaries?,
+          quarantine:     quarantine?,
+          require_sha:    require_sha?,
+          skip_cask_deps: skip_cask_deps?,
+          verbose:        verbose?,
+        )
+      end
+
+      def self.upgrade_casks(
+        *casks,
+        force: nil,
+        greedy: nil,
+        dry_run: nil,
+        binaries: nil,
+        skip_cask_deps: nil,
+        verbose: nil,
+        quarantine: nil,
+        require_sha: nil
+      )
+        # TODO: Handle this in `CLI::Parser`.
+        binaries    = Homebrew::EnvConfig.cask_opts_binaries?    if binaries.nil?
+        dry_run     = Homebrew::EnvConfig.cask_opts_dry_run?     if dry_run.nil?
+        force       = Homebrew::EnvConfig.cask_opts_force?       if force.nil?
+        greedy      = Homebrew::EnvConfig.cask_opts_greedy?      if greedy.nil?
+        quarantine  = Homebrew::EnvConfig.cask_opts_quarantine?  if quarantine.nil?
+        require_sha = Homebrew::EnvConfig.cask_opts_require_sha? if require_sha.nil?
+        verbose     = Homebrew::EnvConfig.cask_opts_verbose?     if verbose.nil?
```

Comment:
This isn't supported in `HOMEBREW_CASK_OPTS` either, we have `HOMEBREW_VERBOSE` for that.